src/WebExploitation/FileUploads/Content-type restriction.md

```diff
@@ -0,0 +1,89 @@
+# File Uploads: Content-type restriction
+
+## Introduction
+In the wild, it's unlikely that you'll find a website that has no protection against file upload attacks like we saw in the previous lab. But just because defenses are in place, that doesn't mean that they're robust. You can sometimes still exploit flaws in these mechanisms to obtain a web shell for remote code execution. 
+
+
+## Table of Contents
+
+- [File Uploads: Unrestricted File Uploads!](#file-uploads-unrestricted-file-uploads)
+  - [Introduction](#Introduction)
+  - [Table of Contents](#table-of-contents)
+  - [How to identify](#how-to-identify)
+  - [Code/tools/websites](#codetoolswebsites)
+    - [Tools](#tools)
+    - [Websites](#websites)
+  - [Sample problem](#sample-problem)
+    - [Problem Statement](#problem-statement)
+    - [Solution](#solution)
+  - [References](#references)
+  - [Conclusion](#conclusion)
+
+
+## How to identify
+
+From a security perspective, the worst possible scenario is when a website allows you to upload server-side scripts, such as PHP, Java, or Python files, and is also configured to execute them as code. This makes it trivial to create your own web shell on the server. 
+
+**Upload Section** :
+Any type of upload can possibly have a upload vulnerability. Make sure you remember this when you encounter a upload section in real life.
+
+
+
+## Code/tools/websites
+
+### Tools
+- **Burp Suite:** A web vulnerability scanner with tools for manual testing.
+
+### Websites
+> https://portswigger.net/web-security/file-upload
+> https://ctf.hackthebox.com/ 
+
+
+## Sample problem
+
+### Problem Statement
+This lab contains a vulnerable image upload function. It attempts to prevent users from uploading unexpected file types, but relies on checking user-controllable input to verify this.
+To solve the lab, upload a basic PHP web shell and use it to exfiltrate the contents of the file /home/carlos/secret.
+
+
+- [Link to the Problem](https://portswigger.net/web-security/file-upload/lab-file-upload-web-shell-upload-via-content-type-restriction-bypass)
+
+### Solution
+Here we have a upload section under My Accounts tab to upload the Avatar.
+We use burpsuite proxy here to interupt the Requests. 
+
+- **Initial Request for JPG**:
+The Content-Type response header may provide clues as to what kind of file the server thinks it has served. If this header hasn't been explicitly set by the application code, it normally contains the result of the file extension/MIME type mapping. 
+  ```
+  Content-Disposition: form-data; name="avatar"; filename="filename.jpg"
+  Content-Type: image/jpeg
+  ``` 
+
+- **Restriction with MIME type**:
+This time when we try to upload this exploit directly...
```

Comment:
~This time when we try to upload this exploit directly...~
Now, if we try to upload a .php file with the following contents: